src/rules/banParensRule.ts

```diff
@@ -0,0 +1,252 @@
+                    const childOrChildren = parentMapping[whichChild];
+                    return Array.isArray(childOrChildren) ?
+                        childOrChildren.indexOf(node) !== -1 :
+                        childOrChildren === node;
+                },
+            };
+        }));
+
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+        if ((isParenthesizedExpression(node) && !parensAreNecessary(node, ctx.sourceFile)) || isParenthesizedType(node)) {
+            const restriction = restrictions.find((r) => r.test(node));
+            if (restriction != undefined) {
+                let replacement = [
+                    Lint.Replacement.deleteFromTo(node.getStart(), node.getStart() + 1),
+                    Lint.Replacement.deleteFromTo(node.getEnd() - 1, node.getEnd()),
+                ];
+                const charBeforeParens = ctx.sourceFile.text[node.getStart() - 1];
+                // Prevent correcting typeof(x) to typeofx, throw(err) to throwerr
+                if (charBeforeParens.match(/\w/) !== null) {
+                    replacement.push(Lint.Replacement.appendText(node.getStart(), " "));
+                }
+                // Don't suggest a fix for a (hopefully rare) pattern where
+                // removing the parentheses would almost always be bad, e.g.
+                // let x = (y = 1, z = 2);
+                if (isParenthesizedExpression(node) &&
+                    isBinaryExpression(node.expression) &&
+                    node.expression.operatorToken.kind === ts.SyntaxKind.CommaToken) {
+                    replacement = [];
+                }
+                ctx.addFailureAtNode(
+                    node,
+                    Rule.FAILURE_STRING_FACTORY(restriction.message),
+                    replacement);
+            }
+        }
+        return ts.forEachChild(node, cb);
+    });
+}
+
+/**
+ * Checks some exceptional cases where the parentheses likely are still required.
+ */
+function parensAreNecessary(node: ts.ParenthesizedExpression, sourceFile: ts.SourceFile) {
+    return (
+        // Don't flag `(0).foo()`, because `0.foo()` doesn't work.
+        (isNumericLiteral(node.expression) &&
+            isPropertyAccessExpression(node.parent!)) ||
+        // Don't flag `return (\nfoo)`, since the parens are necessary.
+        (isReturnStatement(node.parent!) &&
+            !isSameLine(sourceFile, node.getStart(), node.expression.getStart())) ||
+        // Don't flag parens around destructuring assignment
+        (isBinaryExpression(node.expression) &&
+            node.expression.operatorToken.kind === ts.SyntaxKind.EqualsToken &&
+            isObjectLiteralExpression(node.expression.left) &&
+            isExpressionStatement(node.parent!)) ||
+        // Don't flag parentheses in an arrow function's body
+        (isArrowFunction(node.parent!) &&
+            (node.parent as ts.ArrowFunction).body === node));
```

Comment:
you can remove the second part of the condition. A `ParenthesizedExpression` can only occur in the body of an `ArrowFunction`

Reply:
Done.